detector/muonSystem/muonSystemMuRWELL_o1_v01.cpp

```diff
@@ -0,0 +1,941 @@
+
+
+  auto barrelRadiatorParameters = xmlElement.child(_Unicode(barrelRadiatorParameters));
+  int numBarrelRadiators = barrelRadiatorParameters.attr<double>("numBarrelRadiators");
+  double barrelRadiatorThickness = barrelRadiatorParameters.attr<double>("barrelRadiatorThickness");
+  double barrelRadiatorLayerRadius = barrelRadiatorParameters.attr<double>("barrelRadiatorLayerRadius");
+  dd4hep::Material barrelRadiatorMaterial = lcdd.material(barrelRadiatorParameters.attr<std::string>("material"));
+ // double barrelRadiatorLayer2Radius = barrelDetectorParameters.attr<double>("barrelRadiatorLayer2Radius");
+
+  //                    --- Endcap parameters ---
+
+  auto endcapDetectorParameters = xmlElement.child(_Unicode(endcapDetectorParameters));
+  int numEndcapDetectorLayers = endcapDetectorParameters.attr<double>("numEndcapDetectorLayers");
+  double endcapDetectorVolumeThickness = endcapDetectorParameters.attr<double>("endcapDetectorVolumeThickness");
+  double endcapDetectorLayerInnerRadius = endcapDetectorParameters.attr<double>("endcapDetectorLayerInnerRadius");
+  double endcapDetectorLayerOuterRadius = endcapDetectorParameters.attr<double>("endcapDetectorLayerOuterRadius");
+  double endcapDetectorZOffset = endcapDetectorParameters.attr<double>("endcapDetectorZOffset");
+
+
+  auto endcapRadiatorParameters = xmlElement.child(_Unicode(endcapRadiatorParameters));
+  int numEndcapRadiators = endcapRadiatorParameters.attr<double>("numEndcapRadiators");
+  double endcapRadiatorThickness = endcapRadiatorParameters.attr<double>("endcapRadiatorThickness");
+  double endcapRadiatorLayerInnerRadius = endcapRadiatorParameters.attr<double>("endcapRadiatorLayerInnerRadius");
+  double endcapRadiatorLayerOuterRadius = endcapRadiatorParameters.attr<double>("endcapRadiatorLayerOuterRadius");
+  double endcapRadiatorZOffset = endcapRadiatorParameters.attr<double>("endcapRadiatorZOffset");
+  dd4hep::Material endcapRadiatorMaterial = lcdd.material(endcapRadiatorParameters.attr<std::string>("material"));
+
+  // Create layer boxes with their respective material, etc
+  //  auto layers = xmlElement.children(_Unicode(layer));
+  //  auto numLayers = xmlElement.numChildren(_Unicode(layer), true);
+  //  dd4hep::xml::Handle_t layer(layers.reset());
+  //  int sensitiveLayerIndex = 0;
+
+  // -----------------------------------------------------------------------------------------------------------
+
+  double shapeAngle = 360.0 / numSides; // it's the full angle
+  double shapeAngle_radians = M_PI / numSides;  // Isn't it a half angle!!
+  double angle_clearance = 0.0; // 0.02 works good but needs the detector volume thick to be more than 60 mm. // it's less than 1 degree, we use the clearnce to avoid overlapping
+
+  double sideLength = 2 * radius * std::tan(shapeAngle_radians); 
+  double sideLength2 = 2 * (radius + 1.5 * detectorVolumeThickness) * std::tan(shapeAngle_radians);
```

Comment:
What does the 2 signify?

Reply:
It's coming from the formula to calculate the polygon side length `Side Length a. a = 2r tan(π/n)`.

Reply:
If you mean the number 2 in `sideLength2 ` and other variable as well, that represents here the trapezoid's **parallel sides** lengths. just to distinguish between the two sides lengths.

Reply:
Sorry, yes I mean the 2 in `sideLength2`, can you please add what you wrote as a comment.

Reply:
Done, I Have added the comments.